alphabase/smiles/smiles.py

```diff
@@ -0,0 +1,116 @@
+from typing import Optional
+
+import pandas as pd
+from rdkit import Chem
+from rdkit.Chem.rdmolops import ReplaceSubstructs, SanitizeMol
+
+from alphabase.constants.aa import AA_Formula
+from alphabase.constants.modification import MOD_DF
+
+N_TERM_PLACEHOLDER = "[Xe]"
+C_TERM_PLACEHOLDER = "[Rn]"
+
+MOD_DF_SMILES = MOD_DF[MOD_DF["smiles"] != ""]
+
+aa_smiles = {
+    aa: AA_Formula.loc[aa]["smiles"]
+    for aa in AA_Formula.index
+    if not pd.isna(AA_Formula.loc[aa]["smiles"])
+}
+aa_mols = {aa: Chem.MolFromSmiles(aa_smiles[aa]) for aa in aa_smiles}
+
+n_term_modifications = {
+    name: smiles
+    for name, smiles in zip(MOD_DF_SMILES["mod_name"], MOD_DF_SMILES["smiles"])
+    if "@Protein_N-term" in name or "@Any_N-term" in name
+}
+c_term_modifications = {
+    name: smiles
+    for name, smiles in zip(MOD_DF_SMILES["mod_name"], MOD_DF_SMILES["smiles"])
+    if "@Protein_C-term" in name or "@Any_C-term" in name
+}
+ptm_dict = {
+    name: smiles
+    for name, smiles in zip(MOD_DF_SMILES["mod_name"], MOD_DF_SMILES["smiles"])
+    if name not in n_term_modifications and name not in c_term_modifications
+}
+
+
+def validate_correct_args(
+    aa_smiles: str, n_term_mod: Optional[str] = None, c_term_mod: Optional[str] = None
+) -> None:
+    """
+    Validate the amino acid, N-terminal modification, and C-terminal modification.
```

Comment:
Please use Numpy style docstrings in alphaBase :)
https://sphinxcontrib-napoleon.readthedocs.io/en/latest/example_numpy.html